packages/frontend/core/src/modules/app-tabs-header/views/app-tabs-header.tsx

```diff
@@ -28,93 +36,156 @@ import {
   const activeViewIndex = workbench.activeViewIndex ?? 0;
   const onContextMenu = useAsyncCallback(
     async (viewIdx: number) => {
-      await tabsHeaderService.showContextMenu(workbench.id, viewIdx);
+      await tabsHeaderService.showContextMenu?.(workbench.id, viewIdx);
     },
     [tabsHeaderService, workbench.id]
   );
   const onActivateView = useAsyncCallback(
     async (viewIdx: number) => {
-      await tabsHeaderService.activateView(workbench.id, viewIdx);
+      await tabsHeaderService.activateView?.(workbench.id, viewIdx);
     },
     [tabsHeaderService, workbench.id]
   );
   const onCloseTab: MouseEventHandler = useAsyncCallback(
     async e => {
       e.stopPropagation();
 
-      await tabsHeaderService.closeTab(workbench.id);
+      await tabsHeaderService.closeTab?.(workbench.id);
     },
     [tabsHeaderService, workbench.id]
   );
 
+  const { dropTargetRef, closestEdge } = useDropTarget<AffineDNDData>(
+    () => ({
+      closestEdge: {
+        allowedEdges: ['left', 'right'],
+      },
+      onDrop,
+      dropEffect: 'move',
+      canDrop: tabCanDrop(workbench),
+      isSticky: true,
+    }),
+    [onDrop, workbench]
+  );
+
+  const { dragRef, dragging } = useDraggable<AffineDNDData>(
+    () => ({
+      canDrag: dnd,
+      data: {
+        from: {
+          at: 'app-header:tabs',
+          tabId: workbench.id,
+        },
+      },
+      dragPreviewPosition: 'pointer-outside',
+    }),
+    [dnd, workbench.id]
+  );
+
   return (
     <div
-      key={workbench.id}
-      data-testid="workbench-tab"
-      data-active={tabActive}
-      data-pinned={workbench.pinned}
-      className={styles.tab}
+      className={styles.tabWrapper}
+      ref={node => {
+        dropTargetRef.current = node;
+        dragRef.current = node;
+      }}
     >
-      {workbench.views.map((view, viewIdx) => {
-        return (
-          <Fragment key={view.id}>
-            <button
-              key={view.id}
-              data-testid="split-view-label"
-              className={styles.splitViewLabel}
-              data-active={activeViewIndex === viewIdx && tabActive}
-              onContextMenu={() => {
-                onContextMenu(viewIdx);
-              }}
-              onClick={e => {
-                e.stopPropagation();
-                onActivateView(viewIdx);
-              }}
-            >
-              <div className={styles.labelIcon}>
-                {workbench.ready || !workbench.loaded ? (
-                  iconNameToIcon[view.iconName ?? 'allDocs']
-                ) : (
-                  <Loading />
-                )}
-              </div>
-              {workbench.pinned || !view.title ? null : (
-                <div title={view.title} className={styles.splitViewLabelText}>
-                  {view.title}
+      <div
+        key={workbench.id}
+        data-testid="workbench-tab"
+        data-active={tabActive}
+        data-dragging={dragging}
```

Comment:
![CleanShot 2024-08-01 at 18.24.47@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/g3jz87HxbjOJpXV3FPT7/c6be2e91-9a93-495b-a56b-8971b069011a.png)

The draggable will automatically have this attribute